src/Analyzers/CSharp/Analyzers/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -199,19 +187,42 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
 
                     // Check is a reference of any sort (i.e. read/write/nameof) to the local.
                     if (identifierName.Identifier.ValueText == localSymbol.Name)
-                    {
                         return;
-                    }
                 }
             }
 
             if (!Analyzer.CanSafelyConvertToPatternMatching(
-                semanticModel, localSymbol, comparison, operand,
-                localStatement, enclosingBlock, cancellationToken))
+                    semanticModel, localSymbol, comparison, operand,
+                    localStatement, enclosingBlock, cancellationToken))
             {
                 return;
             }
 
+            var comparisonEnclosingBlock = comparison.AncestorsAndSelf().FirstOrDefault(n => n is BlockSyntax);
```

Comment:
Can this scenario have a comment similar to the one starting on line 152?

Reply:
sure :)